snippets/core/system-text-json/csharp/DeserializeNullToNonnullableType.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace SystemTextJsonSamples
+{
+    public class DeserializeNullToNonnullableType
+    {
+        public static void Run()
+        {
+            string jsonString;
+            var wf = WeatherForecastFactories.CreateWeatherForecast();
+
+            var serializeOptions = new JsonSerializerOptions();
+            serializeOptions.WriteIndented = true;
+            serializeOptions.Converters.Add(new DateTimeOffsetNullHandlingConverter());
+            jsonString = JsonSerializer.Serialize(wf, serializeOptions);
+            Console.WriteLine($"JSON with valid Date:\n{jsonString}\n");
+
+            var deserializeOptions = new JsonSerializerOptions();
+            deserializeOptions.Converters.Add(new DateTimeOffsetNullHandlingConverter());
+            wf = JsonSerializer.Deserialize<WeatherForecast>(jsonString, deserializeOptions);
+            wf.DisplayPropertyValues();
+
+            jsonString = @"{""Date"": null,""TemperatureCelsius"": 25,""Summary"":""Hot""}";
+            Console.WriteLine($"JSON with null Date:\n{jsonString}\n");
+
+            // The missing-date JSON deserializes with error if the converter isn't used.
```

Comment:
The other option would be to change your POCO to have a nullable struct that can handle null (i.e. `DateTimeOffset?`).

Reply:
Comment should say `null` date. Not `missing-date`. This scenario is different from the required property scenario below, where the `date` is actually missing (and not null).

Reply:
>The other option would be to change your POCO to have a nullable struct that can handle null (i.e. DateTimeOffset?).

I mention that in the text of the doc.